magenta-lib/src/main/scala/magenta/input/resolver/DeploymentResolver.scala

```diff
@@ -1,38 +1,41 @@
 package magenta.input.resolver
 
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.{NonEmptyList => NEL, Validated, ValidatedNel}
+import cats.syntax.cartesian._
+import cats.syntax.traverse._
+import cats.instances.list._
 import magenta.input.{ConfigError, Deployment, DeploymentOrTemplate, RiffRaffDeployConfig}
 
 object DeploymentResolver {
-  def resolve(config: RiffRaffDeployConfig): List[Either[ConfigError, Deployment]] = {
-    config.deployments.map { case (label, rawDeployment) =>
+  def resolve(config: RiffRaffDeployConfig): ValidatedNel[ConfigError, List[Deployment]] = {
+    config.deployments.traverseU[ValidatedNel[ConfigError, Deployment]] { case (label, rawDeployment) =>
```

Comment:
https://github.com/scala/scala/pull/5102 means that the weird `U` bit will go away in 2.12.